test/e2e/scheduler_predicates.go

```diff
@@ -727,4 +727,565 @@ var _ = framework.KubeDescribe("SchedulerPredicates [Serial]", func() {
+		// scheduled onto it.
+		By("Trying to launch a pod with a label to get a node which can launch it.")
+		podName := "with-label-" + string(util.NewUUID())
+		_, err := c.Pods(ns).Create(&api.Pod{
+			TypeMeta: unversioned.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"security": "S1"},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  podName,
+						Image: "gcr.io/google_containers/pause:2.0",
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(framework.WaitForPodRunningInNamespace(c, podName, ns))
+		pod, err := c.Pods(ns).Get(podName)
+		framework.ExpectNoError(err)
+
+		nodeName := pod.Spec.NodeName
+		defer c.Pods(ns).Delete(podName, api.NewDeleteOptions(0))
+
+		By("Trying to apply a random label on the found node.")
+		k := "e2e.inter-pod-affinity.kubernetes.io/zone"
+		v := "china-e2etest"
+		patch := fmt.Sprintf(`{"metadata":{"labels":{"%s":"%s"}}}`, k, v)
+		err = c.Patch(api.MergePatchType).Resource("nodes").Name(nodeName).Body([]byte(patch)).Do().Error()
+		framework.ExpectNoError(err)
+
+		node, err := c.Nodes().Get(nodeName)
+		framework.ExpectNoError(err)
+		Expect(node.Labels[k]).To(Equal(v))
+
+		By("Trying to launch the pod, now with podAffinity.")
+		labelPodName := "with-podaffinity-" + string(util.NewUUID())
+		_, err = c.Pods(ns).Create(&api.Pod{
+			TypeMeta: unversioned.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name: labelPodName,
+				Annotations: map[string]string{
+					"scheduler.alpha.kubernetes.io/affinity": `
+						{"podAffinity": {
+							"requiredDuringSchedulingIgnoredDuringExecution": [{
+							"labelSelector":{
+								"matchExpressions": [{
+									"key": "security",
+									"operator": "In",
+									"values": ["S1", "value2"]
+								}]
+							},
+							"topologyKey": "` + k + `",
+							"namespaces":[{
+								"metadata": {
+									"name": "` + ns + `"
+								}
+							}]
+						}]
+					 }}`,
+				},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  labelPodName,
+						Image: "gcr.io/google_containers/pause:2.0",
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err)
+		defer c.Pods(ns).Delete(labelPodName, api.NewDeleteOptions(0))
+
+		// check that pod got scheduled. We intentionally DO NOT check that the
+		// pod is running because this will create a race condition with the
+		// kubelet and the scheduler: the scheduler might have scheduled a pod
+		// already when the kubelet does not know about its new label yet. The
+		// kubelet will then refuse to launch the pod.
+		framework.ExpectNoError(framework.WaitForPodNotPending(c, ns, labelPodName))
+		labelPod, err := c.Pods(ns).Get(labelPodName)
+		framework.ExpectNoError(err)
+		Expect(labelPod.Spec.NodeName).To(Equal(nodeName))
+	})
+
+	// test the pod affinity successful matching scenario with diff labels.
```

Comment:
It looks like this test is identical to the previous one except that you do "service" and "S1" instead of "security" and "S1". So I don't think it adds much value. I would suggest replacing this test with one that tests podAntiAffinity. (I don't think any of your e2e tests test podAntiAffinity by itself). You could test it by having a node with a pod with label L, and scheduling a new pod with anti-affinity against L, and checking that the new pod stays pending (using time.Sleep() and verifyResult(), as in the other tests).